### PR TITLE
Size Op - CUDA kernel support

### PR DESCRIPTION
**Pull request by ytaous**
Current Size Op is CPU only. 

This code change is to enable CUDA support as well in order to avoid extra memcpy when connecting Size OP among other CUDA kernel enabled OPs.

During the test of Mainz/T5 model, these extra memcpy introduced by CPU kernel only Size OP is slowing down the training performance.

### CODE REVIEW

File: onnxruntime/core/providers/cuda/tensor/size.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+namespace cuda {
+
+ONNX_OPERATOR_KERNEL_EX(
+    Size,
+    kOnnxDomain,
+    1,
+    kCudaExecutionProvider,
+    KernelDefBuilder()
+        .OutputMemoryType<OrtMemTypeCPUOutput>(0)
+        .TypeConstraint("T",
```
Comment (SherlockNoMad) [inline onnxruntime/core/providers/cuda/tensor/size.cc:17]: .TypeConstraint("T", DataTypeImpl::AllTensorTypes()), 

?

Comment (SherlockNoMad) [reply onnxruntime/core/providers/cuda/tensor/size.cc:17]: MLFloat16 and BFloat16 missing... 
AllTensorTypes should be fine here.

Comment (SherlockNoMad) [inline onnxruntime/core/providers/cuda/tensor/size.cc:17]: .OutputMemoryType<OrtMemTypeCPUOutput>(0)

output should be in CPU

Comment (ytaous) [reply onnxruntime/core/providers/cuda/tensor/size.cc:17]: ok

Comment (ytaous) [reply onnxruntime/core/providers/cuda/tensor/size.cc]: resolved per discussion - pls let me know if I misunderstood

File: onnxruntime/core/providers/cuda/gpu_data_transfer.cc [hunk 2]
```diff
@@ -32,8 +32,14 @@ common::Status GPUDataTransfer::CopyTensor(const Tensor& src, Tensor& dst, int e
 
   if (dst_device.Type() == OrtDevice::GPU) {
     if (src_device.Type() == OrtDevice::CPU && src_device.MemType() == OrtDevice::MemType::CUDA_PINNED) {
-      // copy from pinned memory to GPU, this is non-blocking
+      // copy from pinned memory to GPU
+      // this is blocking for Windows as it's not thread-safe with Async call - ReduceMeanGrad UT with RelWithDebInfo build
+      // is broken consistently 
+#ifdef _WIN32
+      CUDA_RETURN_IF_ERROR(cudaMemcpy(dst_data, src_data, bytes, cudaMemcpyHostToDevice));
+#else
```
Comment (weixingzhang) [inline onnxruntime/core/providers/cuda/gpu_data_transfer.cc:40]: Will Size op trigger this path?

Comment (ytaous) [reply onnxruntime/core/providers/cuda/gpu_data_transfer.cc]: yes - now reverted and make proper fix after discussing with Ke.

Comment (weixingzhang) [reply onnxruntime/core/providers/cuda/gpu_data_transfer.cc]: yes, to me this is a workaround hiding some issue in ORT.

**Pull request merged**